build/package.js

```diff
@@ -0,0 +1,151 @@
+const { resolve } = require('path')
+const { spawnSync } = require('child_process')
+const consola = require('consola')
+const { readFileSync, existsSync, readJSONSync, writeFileSync, copySync } = require('fs-extra')
+const builtin = require('./builtin.json')
+
+module.exports = class Package {
+  constructor(options) {
+    this.options = Object.assign({}, options)
+
+    // Use pwd() as rootDir if not specified
+    if (!this.options.rootDir) {
+      this.options.rootDir = process.cwd()
+    }
+
+    // Try to read package.json if not provided
+    if (!this.packageObj) {
+      this.readPackage()
+    }
+  }
+
+  get packageObj() {
+    return this.options.packageObj
+  }
+
+  set packageObj(packageObj) {
+    this.options.packageObj = packageObj
+  }
+
+  get packagePath() {
+    return resolve(this.options.rootDir, 'package.json')
+  }
+
+  readPackage() {
+    if (existsSync(this.packagePath)) {
+      this.packageObj = readJSONSync(this.packagePath)
+    } else if (!this.packageObj) {
+      this.packageObj = {}
+    }
+  }
+
+  writePakage() {
+    writeFileSync(this.packagePath, JSON.stringify(this.packageObj, null, 2))
+  }
+
+  generateVersion() {
+    const date = Math.round(Date.now() / (1000 * 60))
+    const gitCommit = this._gitShortCommit()
+    const baseVersion = this.packageObj.version.split('-')[0]
+    this.packageObj.version = `${baseVersion}-${date}.${gitCommit}`
+  }
+
+  publish(tag = 'latest') {
+    consola.log(`publishing ${this.packageObj.name}@${this.packageObj.version} with tag ${tag}`)
+    const result = this._exec('npm', `publish --tag ${tag}`)
+    return result
+  }
+
+  copyFieldsFrom(source, fields = []) {
+    for (const field of fields) {
+      this.packageObj[field] = source.packageObj[field]
+    }
+  }
+
+  copyFilesFrom(source, files) {
+    if (!files) {
+      files = source.packageObj.files || []
+    }
+
+    for (const file of files) {
```

Comment:
Widening parenthesis as blocks eliminates the need for extra variables -- since this style is already heavily employed throughout Nuxt, can we also do it for things like this?

```
    for (const file of files) {
      copySync(
        resolve(source.options.rootDir, file),
        resolve(this.options.rootDir, file)
      )
    }
```

Reply:
I've kept variables for reasons:

- No need to add extra comments. The code is much readable when using well-named variables
- Adding complexities (Like more logics for resolving path with some `if` conditions) would require fewer changes to the code
- Primarily I've separated variables soi can add some console logs about current operation without repeating code

Reply:
@pi0 yup, I do that too. And fair enough. Usually though when I realize there are no further references to something, I edit it into the call in the final revision. Perhaps we can selectively do that when it makes sense just before merging.